src/rgw/rgw_rest_swift.h

```diff
@@ -185,6 +185,29 @@ class RGWBulkDelete_ObjStore_SWIFT : public RGWBulkDelete_ObjStore {
   void send_response();
 };
 
+class RGWInfo_ObjStore_SWIFT : public RGWInfo_ObjStore {
+protected:
+  struct info
+  {
+    bool is_admin_info;
+    void (*list_data)(Formatter& formatter,
+                              md_config_t& config,
+                              RGWRados& store);
+  };
+
+  static vector<pair<string, struct info>> swift_info;
+public:
+  RGWInfo_ObjStore_SWIFT() {}
+  ~RGWInfo_ObjStore_SWIFT() {}
+
+  void execute() override;
+  void send_response() override;
+  static void list_swift_data(ceph::Formatter& formatter, md_config_t& config, RGWRados& store);
+  static void list_tempurl_data(ceph::Formatter& formatter, md_config_t& config, RGWRados& store);
+  static void list_slo_data(ceph::Formatter& formatter, md_config_t& config, RGWRados& store);
+  static bool is_expired(const std::string& expires);
```

Comment:
these static functions, along with `struct info` and `swift_info`, probably don't need to be members of `RGWInfo_ObjStore_SWIFT`. consider making them static to rgw_rest_swift.cc so you don't need to qualify their definitions with `RGWInfo_ObjStore_SWIFT::`, and you can make changes without having to recompile everything that includes rgw_rest_swift.h